packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,456 @@
+      );
+      expect(functionMocks.buildResultTemplatesManager).toHaveBeenCalledWith(
+        exampleEngine
+      );
+    });
+
+    it('should dispatch quantic__registerrecommendationtemplates event', async () => {
+      document.addEventListener(
+        'quantic__registerrecommendationtemplates',
+        // @ts-ignore
+        (event) => functionMocks.eventHandler(event.detail),
+        {once: true}
+      );
+      createTestComponent();
+
+      await flushPromises();
+
+      expect(functionMocks.eventHandler).toHaveBeenCalledTimes(1);
+      expect(functionMocks.eventHandler).toHaveBeenCalledWith({
+        registerTemplates: functionMocks.registerTemplates,
+      });
+    });
+
+    it('should register the fields to include', async () => {
+      const expectedFormattedFieldsToInclude = ['field1', 'field2'];
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledTimes(1);
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledWith(
+        exampleEngine
+      );
+      expect(functionMocks.dispatch).toHaveBeenCalledTimes(1);
+      expect(functionMocks.dispatch).toHaveBeenCalledWith(
+        functionMocks.registerFieldsToInclude(expectedFormattedFieldsToInclude)
+      );
+    });
+
+    describe('with custom #numberOfRecommendations option', () => {
+      it('should build the buildRecommendationList controller with the custom number of recommendation value', async () => {
+        createTestComponent({
+          ...defaultOptions,
+          numberOfRecommendations: 5,
+        });
+        await flushPromises();
+
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+          exampleEngine,
+          {
+            options: {
+              id: defaultOptions.recommendation,
+              numberOfRecommendations: 5,
+            },
+          }
+        );
+      });
+    });
+
+    describe('when invalid property values are passed', () => {
+      it('should diplay an error message when recommendationsPerRow is not a number', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          recommendationsPerRow: 'invalid',
+        });
+        await flushPromises();
+
+        const initializationError = element.shadowRoot.querySelector(
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
+      });
+
+      it('should diplay an error message when recommendationsPerRow is a negative number', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          recommendationsPerRow: -1,
+        });
+        await flushPromises();
+
+        const initializationError = element.shadowRoot.querySelector(
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
```

Comment:
You can also check that `console.error` has been called here as well since you are already mocking it!